### PR TITLE
Disallow default header to be overwritten

### PR DESCRIPTION
**Pull request by whynick1**
To resolve https://github.com/apache/iceberg-python/issues/550.

### CODE REVIEW

File: tests/catalog/test_rest.py [hunk 1]
```diff
@@ -277,36 +277,55 @@ def test_properties_sets_headers(requests_mock: Mocker) -> None:
     )
 
     catalog = RestCatalog(
-        "rest", uri=TEST_URI, warehouse="s3://some-bucket", **{"header.Content-Type": "application/vnd.api+json"}
+        "rest",
+        uri=TEST_URI,
+        warehouse="s3://some-bucket",
+        **{"header.Content-Type": "application/vnd.api+json", "header.Customized-Header": "some/value"},
```
Comment (whynick1) [inline tests/catalog/test_rest.py:283]: Ensure custom header is still supported, meanwhile disallow default headers, such as `Content-Type` to be overwritten.

**Comment by whynick1** [comment]
@flyrain @liurenjie1024 Can you take a look please. Thanks.

**Review by flyrain — APPROVED** [review]
+1 Thanks @whynick1 for working on it.

**Comment by flyrain** [comment]
@Fokko, do you wanna take a look?

**Pull request merged**